app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -74,6 +77,12 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
   private fun subscribeToProfileLiveData() {
     getProfileData().observe(fragment, Observer<Profile> {
       navigationDrawerHeaderViewModel.profileName.set(it.name)
+      navigationDrawerFooterViewModel.isAdmin.set(it.isAdmin)
+      binding.administratorControlsLinearLayout.setOnClickListener {
+        val intent = AdministratorControlsActivity.createAdministratorControlsActivityIntent(activity, internalProfileId)
```

Comment:
Control this click listener from viewmodel and a routing interface like `RouteToAdministratorControls`. Checkout `StorySummaryViewModel` for reference.

Reply:
Done